ui/signed-out/src/androidMain/kotlin/social/androiddev/signedout/signin/SignInWebView.kt

```diff
@@ -9,89 +9,97 @@
  */
 package social.androiddev.signedout.signin
 
-import android.graphics.Color
-import android.webkit.CookieManager
-import android.webkit.WebResourceError
-import android.webkit.WebResourceRequest
-import android.webkit.WebStorage
-import android.webkit.WebView
-import android.webkit.WebViewClient
+import android.app.Activity
+import android.content.Intent
+import android.net.Uri
+import android.os.Handler
+import android.os.Looper
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.browser.customtabs.CustomTabsIntent
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.CircularProgressIndicator
+import androidx.compose.material.MaterialTheme
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.viewinterop.AndroidView
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.toArgb
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.dp
+import androidx.core.app.OnNewIntentProvider
+import androidx.core.util.Consumer
 
 @Composable
 actual fun SignInWebView(
     modifier: Modifier,
     url: String,
     onWebError: (message: String) -> Unit,
+    onCancel: () -> Unit,
     shouldCancelLoadingUrl: (url: String) -> Boolean,
 ) {
+    val webIntent = webBrowserIntent(url, MaterialTheme.colors.primary, MaterialTheme.colors.secondary)
+    val handler = Handler(Looper.getMainLooper())
 
-    DisposableEffect(Unit) {
-        onDispose {
-            // Remove user session from WebView
-            WebStorage.getInstance().deleteAllData()
-            CookieManager.getInstance().removeAllCookies(null)
+    val launcher =
+        rememberLauncherForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+            if (result.resultCode == Activity.RESULT_CANCELED) {
```

Comment:
I think you could get a result intent from `result.data` when `result.resultCode == Activity.OK`, right? This could save us some logic with `OnNewIntent` and `handler`

Reply:
It's two separate intents, the `onNewIntent` is the deep link from the oauth call, and the `onActivityResult` is the response of the custom tab closing. I am open to suggestions to make this cleaner though.